File: Sorting Algorithms/Python/selection_sort.py
def selection_sort(arr):
    for sorted_till in range(len(arr)-1): #keeps track of the position before which all elements are sorted
        min_index = sorted_till #taking the first element in the unsorted part as the index with the minimum value
        
        for i in range(sorted_till+1, len(arr)): #iterating from the 2nd value in the unsorted part till the end of the list
            if arr[i] < arr[min_index]: #comparing at each step to get the index carrying the minimum value
                min_index = i #assigning a minimum value index
            
        arr[sorted_till], arr[min_index] = arr[min_index], arr[sorted_till] #placing the minimum value found in the unsorted part of the list to the end of of the sorted part
    return arr
    

def main():
    from random import randint, shuffle
    
    a = [randint(0, 50) for _ in range(10)]
    shuffle(a)
    print("Unsorted:", a, end = "\n\n")
    print("Sorted:", selection_sort(a), end = "\n\n")
    

main()

File: Sorting Algorithms/Java/SelectionSort.java
import java.util.*;

class SelectionSort
{
    public static void selectionSort(int[] arr) {
        int minValueIndex;
        int temp;
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {  // keeps track of the starting position of the unsorted part of the array, i.e the position where the next smallest element should be put
            minValueIndex = i;  // takes the first element of the unsorted array as the minimum until finds a smaller element
            for (int j = i + 1; j < n; j++)  // loops through all elements of the unsorted part of the array to look for the minimum element
                if (arr[j] < arr[minValueIndex])  // checks if a particular element is smaller than the minimum till now
                    minValueIndex = j;  // stores the index of the minimum element till now

            // swapping the minimum element found in the unsorted part of the array with the first element of the unsorted array such that it becomes a part of the sortedd array
            temp = arr[i];
            arr[i] = arr[minValueIndex];
            arr[minValueIndex] = temp;

        }
    }


    public static void main(String[] args) {
        int[] array = {14,37,1,2,45,24,18,49,9};

        selectionSort(array);  // does not return anything but instead modifies the original array, since Selection Sort is an INPLACE sorting algorithm
        System.out.println(Arrays.toString(array));
    }
}
